File: src/lib/api/system.ts
import { invoke } from "@tauri-apps/api/core";
import type { DiskInfo } from "@/types/storage";

export function listDisks(): Promise<DiskInfo[]> {
  return invoke<DiskInfo[]>("list_disks");
}

export function openPath(path: string): Promise<void> {
  return invoke<void>("open_path", { path });
}

export function revealPath(path: string): Promise<void> {
  return invoke<void>("reveal_path", { path });
}

File: src-tauri/src/commands/system.rs
use crate::core::error::AppError;
use crate::system::{self, DiskInfo};

#[tauri::command]
pub fn list_disks() -> Vec<DiskInfo> {
    system::list_disks()
}

#[tauri::command]
pub fn open_path(path: String) -> Result<(), AppError> {
    system::open_path(&path)
}

#[tauri::command]
pub fn reveal_path(path: String) -> Result<(), AppError> {
    system::reveal_path(&path)
}
